### PR TITLE
feat: Prepare buy AddCard Screen

### PR DESCRIPTION
**Pull request by lrupnikk**

### CODE REVIEW

File: brd-android/buy/src/main/java/com/fabriik/buy/ui/addcard/AddCardFragment.kt [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import androidx.lifecycle.lifecycleScope
+import androidx.navigation.fragment.findNavController
+import com.fabriik.buy.R
+import com.fabriik.buy.databinding.FragmentAddCardBinding
+import com.fabriik.common.ui.base.FabriikView
+import kotlinx.coroutines.flow.collect
+
+class AddCardFragment : Fragment(), FabriikView<AddCardContract.State, AddCardContract.Effect> {
+
+    lateinit var binding: FragmentAddCardBinding
```
Comment (roknovak) [inline brd-android/buy/src/main/java/com/fabriik/buy/ui/addcard/AddCardFragment.kt]: Could be private

File: brd-android/buy/src/main/java/com/fabriik/buy/ui/addcard/AddCardFragment.kt [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            AddCardContract.Effect.Dismiss ->
+                activity?.finish()
+
+            AddCardContract.Effect.Confirm -> {}
+        }
+    }
+
+    private fun formatCardField(text: CharSequence) {
+        when {
+            text.length == 5 && text[4].isDigit() -> binding.etCardNumber.text?.insert(4, " ")
```
Comment (roknovak) [inline brd-android/buy/src/main/java/com/fabriik/buy/ui/addcard/AddCardFragment.kt]: Create utils class and use regex expression for formatting. Example function:

```kotlin
    fun formatCardNumber(input: String?): String? {
        if (input.isNullOrEmpty()) {
            return input
        }

        val delimiter = "  "

        // remove all whitespaces from input
        val cardNumberPlain = input.replace("\\s+".toRegex(), "")

        // return formatted input
        return cardNumberPlain.replace(
            ".{4}(?!$)".toRegex(), "$0$delimiter"
        )
    }
```

File: brd-android/buy/src/main/java/com/fabriik/buy/ui/addcard/AddCardFragment.kt [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            text.length == 10 && text[9].isDigit() -> binding.etCardNumber.text?.insert(9, " ")
+            text.length == 15 && text[14].isDigit() -> binding.etCardNumber.text?.insert(14, " ")
+        }
+    }
+
+    private fun formatDate(text: CharSequence) {
+        when {
+            text.length == 3 && text[2].isDigit() -> binding.etDate.text?.insert(2, "/")
```
Comment (roknovak) [inline brd-android/buy/src/main/java/com/fabriik/buy/ui/addcard/AddCardFragment.kt]: Same here, create utils class and use SimpleDateFormatter class.

```kotlin
val formatter = new SimpleDateFormat("dd/MM/yyyy")
return formatter.format(date)
```

Comment (roknovak) [reply brd-android/buy/src/main/java/com/fabriik/buy/ui/addcard/AddCardFragment.kt]: Or use something similar

File: brd-android/buy/src/main/res/layout/fragment_add_card.xml [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        android:layout_height="match_parent"
+        android:fillViewport="true"
+        app:layout_behavior="@string/appbar_scrolling_view_behavior">
+
+        <LinearLayout
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:clipToPadding="false"
+            android:orientation="vertical"
+            android:paddingStart="16dp"
```
Comment (roknovak) [inline brd-android/buy/src/main/res/layout/fragment_add_card.xml]: You can use `paddingHorizontal`

File: brd-android/buy/src/main/java/com/fabriik/buy/ui/Utils.kt [hunk 5]
```diff
@@ -0,0 +1,27 @@
+package com.fabriik.buy.ui
```
Comment (roknovak) [inline brd-android/buy/src/main/java/com/fabriik/buy/ui/Utils.kt]: Should be located in com.fabriik.buy.utils

**Review by roknovak — APPROVED** [review]

**Review by dabitdev — APPROVED** [review]

**Pull request merged**